onnx/defs/function.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Facebook Inc. and Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "onnx/defs/function.h"
+#include "onnx/checker.h"
+
+namespace ONNX_NAMESPACE {
+using namespace checker;
+FunctionBuilder& FunctionBuilder::SetDomain(const std::string& domain) {
+  domain_ = domain;
+  return *this;
+}
+
+const std::string& FunctionBuilder::GetDomain() const {
+  return domain_;
+}
+
+FunctionBuilder& FunctionBuilder::SetBuildFunction(BuildFunction build_func) {
+  build_func_ = build_func;
+  return *this;
+}
+
+BuildFunction FunctionBuilder::GetBuildFunction() const {
+  return build_func_;
+}
+
+Status FunctionBuilderRegistry::Register(
+    const FunctionBuilder& function_builder) {
+  std::lock_guard<std::mutex> lock(mutex_);
+  function_builders.push_back(function_builder);
+  return Status::OK();
+}
+
+// Get functions for specific domain.
+Status FunctionBuilderRegistry::GetFunctions(
```

Comment:
Should there be another version of GetFunction that return the function of specific name and specific version (default to latest)?

Reply:
I'd not do it but having upper layer to do that, because this get function is indeed calling BuildFunctions to create FunctionProto, I don't want to expose any API for doing this creation again and again.

Reply:
the builder function is not exposed. I don't want to do caching here. Upper layer callers can do that.